CPO-13 Dynamic configuration (#336)
hazelcast/hz-docs: docs/modules/configuration/pages/understanding-configuration.adoc

[[checking-configuration]]

Hazelcast members and clients looks for static configuration options in the following order:

[cols="10%a,45%a,45%a"]
|===
|Order|Member|Client

|1
|`Config` object provided by programmatic configuration
|`Config` object provided by programmatic configuration

|2
|`hazelcast.config` system property
|`hazelcast.client.config` system property

|3
|`hazelcast.xml` declarative configuration file in the working directory
|`hazelcast-client.xml` declarative configuration file in the working directory

|4
|`hazelcast.xml` declarative configuration file on the classpath
|`hazelcast-client.xml` declarative configuration file on the classpath (Java)

|5
|`hazelcast.yaml` declarative configuration file in the working directory
|`hazelcast-client.yaml` declarative configuration file in the working directory

|6
|`hazelcast.yaml` declarative configuration file on the classpath
|`hazelcast-client.yaml` declarative configuration file on the classpath (Java)

|===

If Hazelcast does not find any of these configuration options, it loads the default configuration file (`hazelcast-default.xml`), which comes with your Hazelcast package.

== Dynamic Configuration

> Dynamic configuration allows you to add configuration for some data structures at runtime, using the Java API. If you use the `Config` object to configure a member, dynamic configuration cannot be persisted. When a cluster restarts, any changes are not reapplied to the cluster.

Author: ```suggestion
Dynamic configuration allows you to add configuration blocks for some features at runtime.
```
